# To Determine whether the building status influences their ability to conform with disabilty regulations using an independent chi-square

*The first thing that needs to be done is to import packages.  You'll only need pandas and stats from scipy*

## Import Packages

In [2]:
import pandas as pd
from scipy import stats

*Next, you will need to import your data. The data you'll be using today was collected from the U.S. Government about the condition their buildings are in.*

## Import Data

In [3]:
government_buildings = pd.read_csv('C:/Users/meredith.dodd/Documents/Data Science/105 Intermediate Statistics/Lesson 1/government_buildings.csv')

In [4]:
government_buildings.head()

Location Code  Region Code    Bldg Address1    Bldg Address2    Bldg City  \
0        CT0013            1      135 HIGH ST                      HARTFORD   
1        CT0024            1    141 CHURCH ST  U.S. COURTHOUSE    NEW HAVEN   
2        CT0047            1     135 GRAND ST                     WATERBURY   
3        CT0049            1  919 MAIN STREET                   WILLIMANTIC   
4        CT0053            1      450 MAIN ST                      HARTFORD   

  Bldg County Bldg State  Bldg Zip Congressional District Bldg Status  \
0    HARTFORD         CT  61031125                      1      ACTIVE   
1   NEW HAVEN         CT  65102001                      3      ACTIVE   
2   NEW HAVEN         CT  67029998                      5      ACTIVE   
3     WINDHAM         CT  62269998                      2      ACTIVE   
4    HARTFORD         CT  61033022                      1      ACTIVE   

  Property Type  Bldg ANSI Usable  Total Parking Spaces Owned/Leased  \
0      BUILDING             97072                    29        OWNED   
1      BUILDING             69037                     0        OWNED   
2      BUILDING              7442                     2       LEASED   
3      BUILDING               198                     0       LEASED   
4      BUILDING            203358                   103        OWNED   

  Construction Date   Historical Type         Historical Status  \
0          1-Jan-33  Field Not In Use  National Register Listed   
1          1-Jan-19  Field Not In Use  National Register Listed   
2          1-Jan-33  Field Not In Use  National Register Listed   
3          1-Jan-11  Field Not In Use             Not Evaluated   
4          1-Jan-63  Field Not In Use  Evaluated - Not Historic   

  ABA Accessibility Flag   
0            Will Conform  
1            Will Conform  
2            Will Conform  
3            Will Conform  
4            Will Conform

*Next, you will need to create a crosstab, or contingency table, with the two variables you are interested in. Since your initial question was to see how the status of the buildings influences whether they conform with disability / accessibilty requirements, those are what you will specify in your crosstab below.  The table can easily be made by using the function from pandas .crosstab(). The arguments are just the data frame name and variable you are interested in. It's important to make a crosstab first because the code for doing chi-squares in python requires a crosstab, not raw data.* 

In [6]:
buildings_crosstab = pd.crosstab(government_buildings['Bldg Status'], government_buildings['ABA Accessibility Flag '])

*And then, of course, you need to call it to view it. Kind of like calling a dog by name and it comes to you. Here you see the building status on the left side, and the information about disability, with the options of No, Will conform, and yes, along the top.  The numbers in the middle are how many met both conditions.  So, there are 332 active buildings that do not meet disdability accessibility standards, for instance.* 

In [7]:
buildings_crosstab

ABA Accessibility Flag    No  Will Conform   Yes
Bldg Status                                     
ACTIVE                   332          3980  4906
DECOMMISSIONED             0            12     1
EXCESS                     5            45    11

*Now that you have your crosstab, you are ready to utilize the function chi2_contingency() to run your independetnt chi-square. The only argument you need in the parentheses is the name of the crosstab you just created.*

In [8]:
stats.chi2_contingency(buildings_crosstab)

(43.41998286049691,
 8.466179848202505e-09,
 4,
 array([[3.34316186e+02, 4.00484998e+03, 4.87883384e+03],
        [4.71480844e-01, 5.64797675e+00, 6.88054240e+00],
        [2.21233319e+00, 2.65020448e+01, 3.22856220e+01]]))

*And there you are.  The first number you get, 43.42, is the chi-square statistic. Then the second number you get is the p value for the chi-square.  If it is less than .05, then there is a significant influence of building status on conforming with disability accessibility standards.  In this case, your p value is smaller than .05, since this is written in scientific notation.  Your p value is very, very small.*

*The array portion down below shows the expected counts in the cells.  This is what would happen if there was NO relationshipo between the variables. You want all the values to be over 5 in order to meet the assumption for your chi-square. Not all of them are - see the left middle value, which has an expected count of .04? That means that it wasn't really appropriate to run this analysis, and that you should either proceed with things with EXTREME caution or you should try and look at different variables, run a different test, or somehow augment your sample size.* 